agent/utils/persistent_store.py

```
@@ -206,14 +299,26 @@ def save(cls, data: dict):
             with open(tmp_path, "w", encoding="utf-8") as f:
                 json.dump(data, f, ensure_ascii=False, indent=2)
-            shutil.move(tmp_path, path)
+                f.flush()
+                os.fsync(f.fileno())   # 不 fsync,断电后这次 replace 可能落地成空文件
+            # 必须是 os.replace 而不是 shutil.move:后者在 Windows 上目标已存在时
+            # os.rename 会抛 FileExistsError,于是回落到 copy2(先截断再写)——非原子。
+            # 也就是说除首次创建外,过去每一次 save 走的都是非原子路径,
+            # 正好推翻本文件开头"原子写入:防止断电导致文件损坏"那句承诺。
+            os.replace(tmp_path, path)
             return True
         except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)